File: go-sdk/pkg/models/operations/postapireactionstoggle.go
package operations

type PostAPIReactionsToggleCategoryEnum string

const (
	PostAPIReactionsToggleCategoryEnumLike          PostAPIReactionsToggleCategoryEnum = "like"
	PostAPIReactionsToggleCategoryEnumUnicorn       PostAPIReactionsToggleCategoryEnum = "unicorn"
	PostAPIReactionsToggleCategoryEnumExplodingHead PostAPIReactionsToggleCategoryEnum = "exploding-head"
	PostAPIReactionsToggleCategoryEnumRaisedHands   PostAPIReactionsToggleCategoryEnum = "raised_hands"
	PostAPIReactionsToggleCategoryEnumFire          PostAPIReactionsToggleCategoryEnum = "fire"
)

type PostAPIReactionsToggleReactableTypeEnum string

const (
	PostAPIReactionsToggleReactableTypeEnumComment PostAPIReactionsToggleReactableTypeEnum = "Comment"
	PostAPIReactionsToggleReactableTypeEnumArticle PostAPIReactionsToggleReactableTypeEnum = "Article"
	PostAPIReactionsToggleReactableTypeEnumUser    PostAPIReactionsToggleReactableTypeEnum = "User"
)

type PostAPIReactionsToggleQueryParams struct {
	Category      PostAPIReactionsToggleCategoryEnum      `queryParam:"style=form,explode=true,name=category"`
	ReactableID   int32                                   `queryParam:"style=form,explode=true,name=reactable_id"`
	ReactableType PostAPIReactionsToggleReactableTypeEnum `queryParam:"style=form,explode=true,name=reactable_type"`
}

type PostAPIReactionsToggleRequest struct {
	QueryParams PostAPIReactionsToggleQueryParams
}

type PostAPIReactionsToggleResponse struct {
	ContentType string
	StatusCode  int64
}

File: python-sdk/src/foremapi/models/operations/post_api_reactions_toggle.py
import dataclasses
from enum import Enum

class PostAPIReactionsToggleCategoryEnum(str, Enum):
    LIKE = "like"
    UNICORN = "unicorn"
    EXPLODING_HEAD = "exploding-head"
    RAISED_HANDS = "raised_hands"
    FIRE = "fire"

class PostAPIReactionsToggleReactableTypeEnum(str, Enum):
    COMMENT = "Comment"
    ARTICLE = "Article"
    USER = "User"


@dataclasses.dataclass
class PostAPIReactionsToggleQueryParams:
    category: PostAPIReactionsToggleCategoryEnum = dataclasses.field(metadata={'query_param': { 'field_name': 'category', 'style': 'form', 'explode': True }})
    reactable_id: int = dataclasses.field(metadata={'query_param': { 'field_name': 'reactable_id', 'style': 'form', 'explode': True }})
    reactable_type: PostAPIReactionsToggleReactableTypeEnum = dataclasses.field(metadata={'query_param': { 'field_name': 'reactable_type', 'style': 'form', 'explode': True }})
    

@dataclasses.dataclass
class PostAPIReactionsToggleRequest:
    query_params: PostAPIReactionsToggleQueryParams = dataclasses.field()
    

@dataclasses.dataclass
class PostAPIReactionsToggleResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    
